src/account-multicall/LlamaAccountMulticallExtension.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
+
+import {LlamaAccountMulticallStorage} from "src/account-multicall/LlamaAccountMulticallStorage.sol";
+import {LlamaBaseAccountExtension} from "src/common/LlamaBaseAccountExtension.sol";
+import {LlamaUtils} from "src/lib/LlamaUtils.sol";
+
+/// @title Llama Account Multicall Extension
+/// @author Llama (devsdosomething@llama.xyz)
+/// @notice An account extension that can multicall on behalf of the Llama account.
+/// @dev This contract should be delegate-called from a Llama account.
+contract LlamaAccountMulticallExtension is LlamaBaseAccountExtension {
+  /// @dev Struct to hold target data.
+  struct TargetData {
+    address target; // The target contract.
+    uint256 value; // The target call value.
+    bytes data; // The target call data.
+  }
+
+  /// @dev The call did not succeed.
+  /// @param index Index of the target data being called.
+  /// @param revertData Data returned by the called function.
+  error CallReverted(uint256 index, bytes revertData);
+
+  /// @dev Thrown if the target-selector is not authorized.
+  error UnauthorizedTargetSelector(address target, bytes4 selector);
+
+  /// @notice The Llama account multicall storage contract.
+  LlamaAccountMulticallStorage public immutable ACCOUNT_MULTICALL_STORAGE;
+
+  /// @dev Initializes the Llama account multicall extenstion.
+  constructor(LlamaAccountMulticallStorage accountMulticallStorage) {
+    ACCOUNT_MULTICALL_STORAGE = accountMulticallStorage;
+  }
+
+  /// @notice Multicalls on behalf of the Llama account.
+  /// @param targetData The target data to multicall.
+  /// @return returnData The return data from the target calls.
+  function multicall(TargetData[] memory targetData) external onlyDelegateCall returns (bytes[] memory returnData) {
```

Comment:
Should this function emit an event?

Reply:
No I don't think so. It's not changing any storage in that contract. We don't emit any events for the LlamaExecutor or Account.execute either.